Intermittent PHPunit step errors/failures on ProboCI

## Problem/Motivation
We've been seeing inconsistent intermittent failures during the PHPunit step of our ProboCI config recently.

### Describe the bug
Example ProboCI log output:
```
$ sudo -u www-data -E /src/az-quickstart-scaffolding/vendor/bin/fastest -vvv '/src/az-quickstart-scaffolding/vendor/bin/phpunit {} --colors=always --verbose;'
- 9 shuffled test classes into the queue.
- Will be consumed by 40 parallel Processes.
4	1/9	✔	1 s 753 ms	/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_paragraphs/tests/src/Unit/AZVideoEmbedHelperTest.php
5	2/9	✘	4 s 483 ms	/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_core/tests/src/Functional/ConfigOverrideTest.php[5] /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_core/tests/src/Functional/ConfigOverrideTest.phpPHPUnit 8.5.8 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.16
Configuration: /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/phpunit.xml.dist

E                                                                   1 / 1 (100%)

Time: 4.4 seconds, Memory: 8.00 MB

There was 1 error:

1) Drupal\Tests\az_core\Functional\ConfigOverrideTest::testConfigOverride
Undefined index: value

/src/az-quickstart-scaffolding/web/core/includes/install.core.inc:2285
/src/az-quickstart-scaffolding/web/core/includes/install.core.inc:1076
/src/az-quickstart-scaffolding/web/core/includes/install.core.inc:690
/src/az-quickstart-scaffolding/web/core/includes/install.core.inc:565
/src/az-quickstart-scaffolding/web/core/includes/install.core.inc:118
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Test/FunctionalTestSetupTrait.php:285
/src/az-quickstart-scaffolding/web/core/tests/Drupal/Tests/BrowserTestBase.php:579
/src/az-quickstart-scaffolding/web/core/tests/Drupal/Tests/BrowserTestBase.php:403
/src/az-quickstart-scaffolding/vendor/phpunit/phpunit/src/Framework/TestResult.php:691

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
PHP Warning:  file_put_contents(/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/.phpunit.result.cache): failed to open stream: Permission denied in /src/az-quickstart-scaffolding/vendor/phpunit/phpunit/src/Runner/DefaultTestResultCache.php on line 108

7	3/9	✔	1 min 49 s 79 ms	/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_demo/tests/src/Functional/AZDemoContentTest.php
8	4/9	✔	1 min 52 s 496 ms	/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_cas/tests/src/Functional/AzCasTest.php
1	5/9	✔	1 min 52 s 635 ms	/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_core/tests/src/Functional/ClearCacheTest.php
3	6/9	✔	1 min 53 s 221 ms	/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_paragraphs/tests/src/Functional/AZParagraphsTest.php
6	7/9	✔	1 min 54 s 286 ms	/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_core/tests/src/Functional/MonitoringPageTest.php
2	8/9	✔	3 min 23 s 394 ms	/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/themes/custom/az_barrio/tests/src/Functional/AzBarrioTest.php
9	9/9	✔	3 min 29 s 878 ms	/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_cas/tests/src/Functional/AzCasAdminSettingsTest.php

[5] /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_core/tests/src/Functional/ConfigOverrideTest.phpPHPUnit 8.5.8 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.16
Configuration: /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/phpunit.xml.dist

E                                                                   1 / 1 (100%)

Time: 4.4 seconds, Memory: 8.00 MB

There was 1 error:

1) Drupal\Tests\az_core\Functional\ConfigOverrideTest::testConfigOverride
Undefined index: value

/src/az-quickstart-scaffolding/web/core/includes/install.core.inc:2285
/src/az-quickstart-scaffolding/web/core/includes/install.core.inc:1076
/src/az-quickstart-scaffolding/web/core/includes/install.core.inc:690
/src/az-quickstart-scaffolding/web/core/includes/install.core.inc:565
/src/az-quickstart-scaffolding/web/core/includes/install.core.inc:118
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Test/FunctionalTestSetupTrait.php:285
/src/az-quickstart-scaffolding/web/core/tests/Drupal/Tests/BrowserTestBase.php:579
/src/az-quickstart-scaffolding/web/core/tests/Drupal/Tests/BrowserTestBase.php:403
/src/az-quickstart-scaffolding/vendor/phpunit/phpunit/src/Framework/TestResult.php:691

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
PHP Warning:  file_put_contents(/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/.phpunit.result.cache): failed to open stream: Permission denied in /src/az-quickstart-scaffolding/vendor/phpunit/phpunit/src/Runner/DefaultTestResultCache.php on line 108


    ✘ ehm broken tests...
    Time: 3 minutes 29 seconds 900 milliseconds, Memory: 6.00 MiB
$ exit
```

## Proposed resolution
Identify actual problem.  (i.e. Is this being caused by too many parallel test executions overwhelming Probo?)

### Expected behavior
PHPunit step should only fail for actual failed tests.

## Additional context
The `Undefined index: value` statements in the log are probably coming from this Drupal core issue which prevents Drupal from properly throwing exceptions when certain install requirements aren't met / install errors are encountered:
https://www.drupal.org/project/drupal/issues/3163925

--- a/.probo.yaml
+++ b/.probo.yaml
@@ -59,7 +59,7 @@ steps:
       - export SIMPLETEST_BASE_URL="http://localhost"
       - export SIMPLETEST_DB="mysql://${TEST_DATABASE_USER}:${TEST_DATABASE_PASS}@localhost/${TEST_DATABASE_NAME}"
       - cd /var/www/html/profiles/custom/az_quickstart
-      - sudo -u www-data -E find $SRC_DIR/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/ -name "*Test.php" | sudo -u www-data -E $SRC_DIR/az-quickstart-scaffolding/vendor/bin/fastest -vvv "$SRC_DIR/az-quickstart-scaffolding/vendor/bin/phpunit {} --colors="always" --verbose;"
+      - sudo -u www-data -E find $SRC_DIR/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/ -name "*Test.php" | sudo -u www-data -E $SRC_DIR/az-quickstart-scaffolding/vendor/bin/fastest -p 10 -vvv "$SRC_DIR/az-quickstart-scaffolding/vendor/bin/phpunit {} --colors="always" --verbose --do-not-cache-result;"
   - name: Log in to terminus
     command: 'source ${ASSET_DIR}/terminus.sh; terminus auth:login --machine-token=${TERMINUS_TOKEN}'
   - name: Migration Tests

--- a/composer.json
+++ b/composer.json
@@ -114,7 +114,8 @@
                 "Allow editing area in Claro to span full width": "https://www.drupal.org/files/issues/2020-12-16/3184667-8.patch",
                 "Editing menus user-experience issue (#2957953)": "https://git.drupalcode.org/project/drupal/-/commit/68b026906b7aad748b2427caa52527436a99eb8c.diff",
                 "Unable to select single view display for view settings": "https://www.drupal.org/files/issues/2020-11-03/2552541-43.patch",
-                "MediaEmbed conflates default view mode with user-selected view mode": "https://www.drupal.org/files/issues/2020-11-11/drupal_core-mediaembed_default_view_mode-3109289-24.patch"
+                "MediaEmbed conflates default view mode with user-selected view mode": "https://www.drupal.org/files/issues/2020-11-11/drupal_core-mediaembed_default_view_mode-3109289-24.patch",
+                "Unrelated error message when running tests with database errors (3163925)": "https://git.drupalcode.org/project/drupal/-/merge_requests/212.diff"
             },
             "drupal/config_distro": {
                 "fnmatch issue": "https://www.drupal.org/files/issues/2020-07-04/3144145-replace-fnmatch-with-preg-match-6.patch",

